File: app/src/main/java/com/arbaelbarca/listmovieskatalog/presenter/listmoviespresenter/ListMoviesPresenterImpl.java
package com.arbaelbarca.listmovieskatalog.presenter.listmoviespresenter;

import android.content.Context;
import android.util.Log;

import com.arbaelbarca.listmovieskatalog.model.movies.ResponseMoviesList;
import com.arbaelbarca.listmovieskatalog.network.NetworkApi;
import com.arbaelbarca.listmovieskatalog.view.MainContract;

import rx.Subscriber;
import rx.android.schedulers.AndroidSchedulers;
import rx.schedulers.Schedulers;

public class ListMoviesPresenterImpl implements MainContract.GetNoticeIntractor, MainContract.presenter {

    private Context context;

    public ListMoviesPresenterImpl(Context context) {
        this.context = context;
    }


    @Override
    public void onDestroy() {

    }

    @Override
    public void requestFromDataServer() {

    }

    @Override
    public void refreshData(String textType) {
        getNoticeArrayList((OnFinishedListener) this, textType);

    }

    @Override
    public void getNoticeArrayList(OnFinishedListener onFinishedListener, String textType) {
        NetworkApi
                .getInstance()
                .getAPI()
                .getMovies(textType, "en-US")
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(new Subscriber<ResponseMoviesList>() {
                    @Override
                    public void onCompleted() {
                        Log.d("responEror", " completed");

                    }

                    @Override
                    public void onError(Throwable e) {
                        onFinishedListener.onFailure(e);

                    }

                    @Override
                    public void onNext(ResponseMoviesList responseMoviesList) {
                        if (responseMoviesList.getResults() != null) {
                            onFinishedListener.onFinished(responseMoviesList.getResults());
                            Log.d("responEror", " success ");

                        }
                    }
                });

    }


}

File: app/src/main/java/com/robytakwa/listpicturekatalog/presenter/listmoviespresenter/ListMoviesPresenterImpl.kt
package com.robytakwa.listpicturekatalog.presenter.listmoviespresenter

import android.content.Context
import android.util.Log
import com.robytakwa.listpicturekatalog.model.movies.ResponseMoviesList
import com.robytakwa.listpicturekatalog.network.NetworkApi
import com.robytakwa.listpicturekatalog.utils.Constants.LANGUAGE
import com.robytakwa.listpicturekatalog.view.MainContract
import rx.Subscriber
import rx.android.schedulers.AndroidSchedulers
import rx.schedulers.Schedulers

class ListMoviesPresenterImpl(private val context: Context) : MainContract.GetNoticeIntractor, MainContract.presenter {


    override fun onDestroy() {

    }

    override fun requestFromDataServer() {

    }

    override fun refreshData(textType: String) {
        getNoticeArrayList(this as MainContract.GetNoticeIntractor.OnFinishedListener, textType)

    }

    override fun getNoticeArrayList(onFinishedListener: MainContract.GetNoticeIntractor.OnFinishedListener, textType: String) {
        NetworkApi
                .instance
                .api
                .getMovies(textType, LANGUAGE)
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(object : Subscriber<ResponseMoviesList>() {
                    override fun onCompleted() {
                        Log.d("responEror", " completed")

                    }

                    override fun onError(e: Throwable) {
                        onFinishedListener.onFailure(e)

                    }

                    override fun onNext(responseMoviesList: ResponseMoviesList) {
                        if (responseMoviesList.results != null) {
                            onFinishedListener.onFinished(responseMoviesList.results)
                            Log.d("responEror", " success ")

                        }
                    }
                })

    }


}
